### PR TITLE
Expand precipitation table to include ice and snow, and add precipitation forecast chart

### LINKED ISSUE #1640: Visualization/table: Precip amount, 6-hours, bar
[Figma documentation for tables](https://www.figma.com/design/w2QI2uTgaTWUPLHLi40mm1/Weather.gov-%E2%80%93-v0.2?node-id=12249-169526&t=lOSmXKrhbxj6Vr00-1)

[Figma documentation for charts](https://www.figma.com/design/w2QI2uTgaTWUPLHLi40mm1/Weather.gov-%E2%80%93-v0.2?node-id=12319-175435&t=lOSmXKrhbxj6Vr00-1)

[In Context](https://www.figma.com/design/w2QI2uTgaTWUPLHLi40mm1/Weather.gov-%E2%80%93-v0.2?node-id=12078-134318&m=dev&t=lOSmXKrhbxj6Vr00-1)

Considerations:
- Blocked by #789 
- Blocked by #987 
- Implement within Hourly Details expansion for now, while we wait on #1616

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

- modifies the precipitation periods structure to include liquid, snow, and ice (instead of just `value`)
- adds snow and/or ice to the precip table, if they have any non-zero values
- adds a chart for the precip data, only including the datasets that have some non-zero values
- creates a fill pattern from image
- closes #1640

## What does the reviewer need to know? 🤔

As this is the last PR that should make changes to our PHP backend/data code... I mean, personally I don't feel like maintainability is important anymore since this code will literally exist for like, three weeks. But whoever is reviewing, you're the reviewer, so you choose. 😅 

### Mobile
image

### Desktop
image

### CODE REVIEW

**Comment by partly-igor** [comment]
Visually I think this is looking really good. Just a few notes:
1. The pattern for snow doesn't seem to match in the legend between the chart. In the legend in looks like circles, but in the chart it's rectangles. This could be just a rendering issue – in which case we don't have to spend too much time on it. But wanted to mention in case there is a simple fix.
2. The ice and water bars are missing their stroke
3. On desktop the explainer should be side by side with the table. 
4. Is possible it would get the diagonal stripes to be a lighter color on the ice bar. 

Screenshot of the PR chart with the annotation list

Screenshot of different pattern for snow in the legend and bar



Screenshot of intended pattern on ice bar

File: web/themes/new_weather_theme/assets/js/charts/qpf.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+const chartContainers = Array.from(
+  document.querySelectorAll(".wx-qpf-chart-container"),
+);
+
+const makePattern = async (imageUrl, size = 60) =>
+  new Promise((resolve) => {
+    const imgage = new Image();
+    imgage.src = imageUrl;
+    imgage.onload = () => {
```
Comment (eric-gade) [inline web/themes/new_weather_theme/assets/js/charts/qpf.js:15]: I'm imgagining that was a typo
```suggestion
    const image = new Image();
    image.src = imageUrl;
    image.onload = () => {
```

File: web/themes/new_weather_theme/assets/js/charts/qpf.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      const imageCanvas = document.createElement("canvas");
+      imageCanvas.width = size;
+      imageCanvas.height = size;
+
+      const imageContext = imageCanvas.getContext("2d");
+      imageContext.drawImage(imgage, 0, 0, size, size);
+
+      const pattern = imageContext.createPattern(imageCanvas, "repeat");
```
Comment (eric-gade) [inline web/themes/new_weather_theme/assets/js/charts/qpf.js:23]: This is cool, I had not encountered it before!

**Review by eric-gade — APPROVED** [review]
👍 
## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

File: web/themes/new_weather_theme/assets/js/charts/qpf.js (hunk 1 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/assets/js/charts/qpf.js:15]: Don't let your imgagination run away with you.

**Pull request merged**